module/VuFind/tests/unit-tests/src/VuFindTest/Config/ConfigTest.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * Configuration Wrapper Test Class
+ *
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+
+namespace VuFindTest\Config;
+
+use VuFind\Config\Config;
+
+/**
+ * Configuration Wrapper Test Class
+ *
+ * @category VuFind
+ * @package  Tests
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:testing:unit_tests Wiki
+ */
+class ConfigTest extends \PHPUnit\Framework\TestCase
+{
+    /**
+     * Test configuration data
+     *
+     * @var array
+     */
+    protected static $configArray = [
+        'section1' => [
+            'setting1' => 'value1',
+            'setting2' => 'value2',
+        ],
+        'section2' => [
+            'setting3' => ['foo', 'bar'],
+        ],
+    ];
+
+    /**
+     * Data provider for testBasicBehavior
+     *
+     * @return array[]
+     */
+    public static function basicBehaviorProvider(): array
+    {
+        return [
+            'toArray returns original array' =>
+                [fn ($test, $config) => $test->assertEquals(self::$configArray, $config->toArray())],
```

Comment:
I think it would be more clear to simply add these assertions to the test method without a data provider.

Reply:
I did this so that each would be counted as a separate test, since they don't depend on each other... but if you think making it more readable is preferable, I can refactor.

Reply:
I'd value readability, but I'm okay keeping it this way.

Reply:
I'll approve since everything else looks good to me and let you make the final decision about this. :)

Reply:
Thanks! As we discussed on Slack, I just went ahead and broke out the test class into separate methods... it's more verbose, but it's much more readable. :-)